src/components/sidebar/SafeList/index.tsx

```diff
@@ -75,27 +79,59 @@ const SafeList = ({ closeDrawer }: { closeDrawer?: () => void }): ReactElement =
   const hasNoSafes = Object.keys(ownedSafes).length === 0 && Object.keys(addedSafes).length === 0
   const isWelcomePage = router.pathname === AppRoutes.welcome
 
+  const handleFetchQueued = async () => {
+    const fetchedQueuedTxs: Record<string, string | undefined> = {}
+
+    for (let [chainId, safes] of Object.entries(addedSafes)) {
+      for (let safeAddress of Object.keys(safes)) {
+        const result = await getTransactionQueue(chainId, safeAddress)
+
+        if (result.results.length === 0) continue
+
+        fetchedQueuedTxs[safeAddress] = `${result.results.filter(isTransactionListItem).length}${
+          result.next ? '+' : ''
+        }`
+      }
+    }
+
+    setSafeQueuedTxs(fetchedQueuedTxs)
+  }
+
   return (
     <div className={css.container}>
-      <div className={css.header}>
-        <Typography variant="h4" display="inline" fontWeight={700}>
-          My Safes
-        </Typography>
-        {!isWelcomePage && (
-          <Track {...OVERVIEW_EVENTS.ADD_SAFE}>
-            <Link href={{ pathname: AppRoutes.welcome }} passHref>
-              <Button
-                disableElevation
-                size="small"
-                variant="outlined"
-                onClick={closeDrawer}
-                startIcon={<SvgIcon component={AddIcon} inheritViewBox fontSize="small" />}
-              >
-                Add
-              </Button>
-            </Link>
-          </Track>
-        )}
+      <div className={css.headerWrapper}>
+        <div className={css.header}>
+          <Typography variant="h4" display="inline" fontWeight={700}>
+            My Safes
+          </Typography>
+          <div className={css.headerButtons}>
+            <Button
+              disableElevation
+              size="small"
+              variant="outlined"
+              onClick={handleFetchQueued}
+              startIcon={<SvgIcon component={UpdateIcon} inheritViewBox fontSize="small" />}
+              sx={{ color: ({ palette }) => palette.info.main }}
+            >
+              Update
```

Comment:
Did we explore other namings instead of "Update"? Imo it is not clear what it will do. I've seen other apps call it "Sync addresses" for example.

Reply:
I will raise this with @johannesmoormann to think of a better name that doesn't blow the button size

Reply:
We could poll when the sidebar opens instead of requiring user input (if the last poll after a defined period to prevent spam).